src/Microsoft.AspNetCore.Dispatcher/DispatcherMiddleware.cs

```diff
@@ -1,60 +1,93 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using System.Collections.Generic;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.Routing.Template;
+using Microsoft.Extensions.Options;
 
 namespace Microsoft.AspNetCore.Dispatcher
 {
     public class DispatcherMiddleware
     {
+        private readonly DispatcherOptions _options;
         private readonly RequestDelegate _next;
 
-        public DispatcherMiddleware(RequestDelegate next)
+        public DispatcherMiddleware(IOptions<DispatcherOptions> options, RequestDelegate next)
         {
+            _options = options.Value;
             _next = next;
         }
 
         public async Task Invoke(HttpContext httpContext)
         {
-            var dictionary = new Dictionary<string, DispatcherFeature>
+            foreach (var entry in _options.DispatcherEntryList)
             {
+                var parsedTemplate = entry.RouteTemplate;
+                var defaults = GetDefaults(parsedTemplate);
+                var templateMatcher = new TemplateMatcher(parsedTemplate, defaults);
+                var values = new RouteValueDictionary();
```

Comment:
This doesn't get used, it gets overwritten pretty much immediately.

Reply:
Just stick to `out var` at line 34

Reply:
But it isn't an out parameter

Reply:
OH. You're right by golly. You need to clear it then if the route matches, but none of the endpoints do. TryMatch will dirty the dictionary